### PR TITLE
NetSim: Lobby Hints

### PR DESCRIPTION
**Pull request by islemaster**
Contextual setup help for NetSim levels, behind experiment flag `netsimLobbyHints`.

Adds the following conveniences:
# Hint for signed-out students

screenshot from 2016-08-23 17-55-34

This hint is always displayed to users when providing a display name, which only happens for users that are signed out (not the usual use case for the simulator).
# Hint for students in multiple sections

screenshot from 2016-08-23 16-13-14

This hint is always displayed when selecting from among multiple sections - this screen has needed instructions for a while.
# Hint for first/only student on a shard

screenshot from 2016-08-23 18-04-46

The section name is now always displayed at the top of the lobby.

This hint is displayed beneath the lobby list when the current user is the only user connected to the shard (which tends to happen when students are unable to connect due to user error/setup error, rather than a technical problem).  It vanishes as soon as a second user joins the shard.

The content of this hint is somewhat dynamic.
- The URL under the second bullet point is populated from `window.location`, which has the added benefit that Safari screenshots will now show us which level students are having trouble with.
- The third bullet point is only shown if the student was presented with a choice of sections, as above.  It also contains a convenient `location.reload()` link, since it's non-obvious how to go back and select another section.

### CODE REVIEW

File: apps/src/netsim/NetSimRemoteNodeSelectionPanel.html.ejs [hunk 1]
```diff
@@ -299,6 +304,42 @@ function writeNodeRow(row, nodeStatus, buttonType, addlClass) {
...
+      </p>
+      <ul>
+        <li><strong>Are you signed in?</strong>
+        To use the simulator with your class, you must be
+        signed in to a Code Studio account that is in the same class section as
+        the other students.  Have your teacher confirm that your account is
+        in the class section.</li>
+        <li><strong>Are you on the right level?</strong>
+        You'll only be able to see other students in your class if you are on the
+        same simulator level.  Make sure everyone is on <%= window.location %></li>
```
Comment (Hamms) [inline apps/src/netsim/NetSimRemoteNodeSelectionPanel.html.ejs]: nit: it looks a bit strange to print the url as plain text. Could we do something like wrap it in a `<pre>`?

File: apps/src/netsim/NetSimRemoteNodeSelectionPanel.js [hunk 2]
```diff
@@ -345,6 +360,23 @@ NetSimRemoteNodeSelectionPanel.prototype.shouldShowNode = function (node) {
 };
 
 /**
+ * @returns {boolean} TRUE if the current user is the only client node connected
+ *          to the shard right now.
+ */
+NetSimRemoteNodeSelectionPanel.prototype.isUserAlone = function () {
+  const otherUserCount = this.nodesOnShard_.filter(node =>
```
Comment (Hamms) [inline apps/src/netsim/NetSimRemoteNodeSelectionPanel.js]: could also use [`Array.find`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find), which has the minor advantage of returning as soon as a match is found.

**Comment by Hamms** [comment]
:+1: for friendly, contextually-aware feedback!

LGTM

File: apps/src/netsim/NetSimRemoteNodeSelectionPanel.js (hunk 2 above)
Comment (islemaster) [reply apps/src/netsim/NetSimRemoteNodeSelectionPanel.js]: :facepalm: Replaced with

``` javascript
  return !this.nodesOnShard_.some(node => 
      !this.isMyNode(node) && node.getNodeType() === NodeType.CLIENT);
```

**Pull request merged**